### ISSUE TITLE
Misleading documentation potentially (RailsEventStore::RSpec)?

### ISSUE THREAD

**Issue by fidel** [RailsEventStore/rails_event_store#122]
Posted originally by @paneq 

It is sometimes important to ensure no additional events have been published. Luckliy there's a modifier to cover that usecase.

```ruby
expect(event_store).not_to have_published(an_event(OrderPlaced)).once
expect(event_store).to have_published(an_event(OrderPlaced)).exactly(2).times
```

That's what the documentation says. But...

But looking at the implementation 

```
      def matches_count(events, expected, count)
        return true unless count
        raise NotSupported if expected.size > 1

        expected.all? do |event_or_matcher|
          events.select { |e| event_or_matcher === e }.size.equal?(count)
        end
      end
```

it seems that `once` does not guarantee that _no additional events have been published_, just that the listed events have been published once.

It seems that the author wanted something similar to `.only` proposed in #5 ?

**Comment by fidel** [comment]
@paneq the intention of this matcher is to check whether given event has been published exactly n number of times. `.once` is just a syntax sugar for `exactly(1).times` and it doesn't exclude other events from being published. `.only` could be verifying whether no other events are present.

**Comment by paneq** [comment]
@fidel I understand what the modifier does. What I am saying is that in my opinion the documentation is misleading saying _It is sometimes important to ensure no additional events have been published_. When I read this I expect behavior such as described for `only` which you explained as _could be verifying whether no other events are present_. `once` does not very no additional events have published, but that those specified events have been published limited amount of times.

we could:

* remote the `once` modifier
    * Personally, I don't find it very useful
* change the wording describing what `once` does exactly. 

P.S.

The documentation mentions

```
expect(event_store).to have_published(an_event(OrderPlaced)).exactly(2).times
```

does this work? I haven't seen anything in the code supporting `exactly(2).times`. and `once` is limited to `1` only:

```
      def matches_count(events, expected, count)
        return true unless count
        raise NotSupported if expected.size > 1
```

**Comment by swistak35** [comment]
I've decided to change the documentation, as indeed wording was a little bit off. As it seems that we all agree that `once` should not behave as `only`, then it did not make sense to change that behaviour. Also I ensured that we do support `exactly(2).times` -- we do, `NotSupported` error was only relevant when sometimes use the code like this:

```
expect(event_store).to have_published(an_event(OrderPlaced), an_event(FooEvent).exactly(2).times
```

(note that two different events are passed)

In that case it's not clear at all what should happen, so we do not support that usecase.
